File: src/0006-ZigZag-Conversion/0006.py
class Solution:
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """

        if numRows == 1 and len(s) < numRows:
            return s

        L = [''] * numRows
        index, step = 0, 1

        for x in s:
            L[index] += x
            if index == 0:
                step = 1
            elif index == numRows -1:
                step = -1
            index += step

        return ''.join(L)

File: src/0006-ZigZag-Conversion/0006.cpp
#include <string>
#include <vector>
using namespace std;
static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();

class Solution 
{
public:
    string convert(string s, int numRows) 
    {
        if (numRows == 1 or s.size() < numRows) return s;
        vector<string> L(numRows, "");
        int index = 0, step = 1;
        for (auto ch : s)
        {
            L[index] += ch;
            if (index == 0) step = 1;
            else if (index == numRows - 1) step = -1;
            index += step;
        }
        string res;
        for (auto& s : L) res += s;
        return res;
    }
};